src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -1340,6 +1341,94 @@ public bool WaitForExit(int milliseconds)
             return exited;
         }
 
+        /// <summary>
+        /// Instructs the Process component to wait for the associated process to exit, or
+        /// for the <paramref name="cancellationToken"/> to be canceled.
+        /// </summary>
+        /// <remarks>
+        /// Calling this method will set <see cref="EnableRaisingEvents"/> to true.
+        /// </remarks>
+        public async Task WaitForExitAsync(CancellationToken cancellationToken = default)
+        {
```

Comment:
There's enough expensive work happening as part of this, I suggest adding:
```C#
cancellationToken.ThrowIfCancellationRequested();
```

Reply:
Added!

Reply:
Added!